### ISSUE TITLE
KM Derivations without show lines are correct?

### ISSUE THREAD

**Issue by dsanson** [Carnap/Carnap#46]
A student did all of his direct derivations in Chapter 3 without ever entering show lines. I was a bit surprised that the proofchecker allows this?

<img width="509" alt="1580489816" src="https://user-images.githubusercontent.com/14331/73558357-a2d51980-4418-11ea-8489-7abec09bd8d4.png">

**Comment by gleachkr** [comment]
Yeah, this is the intended behavior at the moment. I've found it useful to be able to really minimize the number of concepts my students need to take in while getting started. So, when I talk through this in class, I sometimes start from the idea of a proof as a series of assertions, beginning from what you know and ending with some new thing you've discovered, and omit the idea of a show line until the second lecture. But it is a little more permissive than you find in the standard KM presentations, I think.

It might work to require an initial show line, and leave the more permissive behavior available behind an option. Or it might work to just have some indication about this way of doing things in the book or documentation.

Any pedagogical preference here?

**Comment by dsanson** [comment]
Ah, that makes lots of sense.

I'm not sure what I think pedagogically. I'll think about it more. My initial reaction: when things get more complicated, in later chapters, I don't think I want students forgoing their initial show line, even if the derivation is going to be direct. So, at a minimum, I think I want the option to turn the permissive behavior off.

But that also makes me wonder whether the entire first chapter on derivations should forgo show lines and DD altogether, so that, in the next chapter, DD and CD can get introduced together, when the contrast makes sense, along with show lines and boxes, which can then be justified as a way of keeping track of assumptions.

But, like I say, I'm not sure! I'm still working out how to adapt my pedagogical approaches to the new degrees of freedom offered by Carnap.
